packages/engine-client/src/utils/queryDataMappers.ts

```diff
@@ -255,3 +259,25 @@ export function mapEngineMarketPrice(
     productId: baseResponse.product_id,
   };
 }
+
+export function mapEngineServerDepositRates(
+  baseResponse: EngineServerMinDepositRatesResponse,
+): GetEngineMinDepositRatesResponse {
+  const minDepositRates: Record<number, EngineMinDepositRate> = mapValues(
+    baseResponse.min_deposit_rates,
+    mapEngineMinDepositRate,
+  );
+
+  return {
+    minDepositRates,
+  };
+}
+
+export function mapEngineMinDepositRate(
```

Comment:
nit: we might be able to just do this inline in `EngineQueryClient`, these fns aren't used / relevant anywhere else

Reply:
gotcha, sg will clean it up